p2p/libp2p/networksharding/listsSharder.go

```diff
@@ -243,7 +257,11 @@ func (ls *listsSharder) splitPeerIds(peers []peer.ID) map[int]sorting.PeerDistan
 		case core.ValidatorPeer:
 			peerDistances[intraShardValidators] = append(peerDistances[intraShardValidators], pd)
 		case core.ObserverPeer:
-			peerDistances[intraShardObservers] = append(peerDistances[intraShardObservers], pd)
+			if peerInfo.PeerSubType == core.FullHistoryObserver {
```

Comment:
As discussed, I think this condition needs to be changed to something like:
```GO
shouldAppendToFullHistory := peerInfo.PeerSubType == core.FullHistoryObserver && ls.maxFullHistoryObservers > 0
if shouldAppendToFullHistory {
////.....

```

Reply:
Done.